M2/Macaulay2/e/coeffrings.hpp

```diff
@@ -66,6 +66,15 @@ class CoefficientRingZZp : public M2::SimpleARing<CoefficientRingZZp>
     if (a < 0) a += p;
     result = log_table[a];
   }
+  
+  void set_from_mpz(elem &result, mpz_t a) const
+  {
+    mpz_t tmp;
+    mpz_init_set_si(tmp, p);        // Convert int p to mpz_t
+    mpz_mod(a, a, tmp);             // a = a mod p (always non-negative)
```

Comment:
Maybe we should keep an mpz_t characteristic here too, so this doesn't need to be done each time?